core/common/src/main/java/alluxio/wire/MasterInfo.java

```diff
@@ -39,6 +41,40 @@
    */
   public MasterInfo() {}
 
+  /**
+   * Creates a new instance of {@link MasterInfo} from a thrift representation.
+   *
+   * @param masterInfo the thrift representation of alluxio master information
+   */
+  protected MasterInfo(alluxio.thrift.MasterInfo masterInfo) {
+    mConfMasterNum = masterInfo.getConfMasterNum();
+    mConfWorkerNum = masterInfo.getConfWorkerNum();
+    mMasterAddress = masterInfo.getMasterAddress();
+    mRpcPort = masterInfo.getRpcPort();
+    mSafeMode = masterInfo.isSafeMode();
+    mStartTimeMs = masterInfo.getStartTimeMs();
+    mUpTimeMs = masterInfo.getUpTimeMs();
+    mVersion = masterInfo.getVersion();
+    mWebPort = masterInfo.getWebPort();
+    mZookeeperAddresses = masterInfo.getZookeeperAddresses();
+  }
+
+  /**
+   * @return the number of masters whose configuration is recorded
```

Comment:
Could we generalize these to return `List<MasterNetAddress>` and `List<WorkerNetAddress>`, and include all live masters/workers? That's enough information for the test, and it's more likely to be useful in the future.

Reply:
Changed to 
```
struct MasterInfo {
  1: string leaderMasterAddress
  2: list<common.MasterNetAddress> masterAddresses
  3: i32 rpcPort
  4: bool safeMode
  5: i64 startTimeMs
  6: i64 upTimeMs
  7: string version
  8: i32 webPort
  9: list<common.MasterNetAddress> workerAddresses
  10: list<string> zookeeperAddresses // Null means zookeeper is not enabled
}
```
but the master addresses and worker addresses are those recorded in server configuration store other than in meta master for the test purpose.
Thanks!